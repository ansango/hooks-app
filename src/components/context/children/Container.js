
import "./Container.css";
export const Container = ({ title = "", children }) => {
  return (
    <div className="base-container py-5 bg-light">
      <h3 className="text-center">{title}</h3>
      <div className="px-2">{children}</div>
    </div>
  );
};
