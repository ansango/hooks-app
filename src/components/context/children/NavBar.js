import { NavLink } from "react-router-dom";

export const NavBar = ({ routes }) => {
  return (
    <nav className="navbar navbar-light bg-light d-flex flex-row justify-content-end">
      <div className="navbar-nav">
        <div className="d-flex">
          {routes.map((route) => (
            <NavLink
              key={route.name}
              exact
              activeClassName="active"
              className="nav-item nav-link mx-3"
              to={route.path}
            >
              {route.name}
            </NavLink>
          ))}
        </div>
      </div>
    </nav>
  );
};
