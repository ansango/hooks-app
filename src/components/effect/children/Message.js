import { useEffect, useState } from "react";

export const Message = () => {
  const [coords, setCoords] = useState({ x: 0, y: 0 });
  const { x, y } = coords;
  useEffect(() => {
    const mouseMove = (e) => setCoords({ x: e.x, y: e.y });
    window.addEventListener("mousemove", mouseMove);
    return () => {
      window.removeEventListener("mousemove", mouseMove);
    };
  }, []);
  return (
    <div>
      X: <p>{x}</p>
      Y: <p>{y}</p>
    </div>
  );
};
