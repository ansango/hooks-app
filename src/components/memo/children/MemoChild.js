import React from "react";

export const MemoChild = React.memo(({ number, increment }) => {
  console.log("render");

  return (
    <button className="btn btn-primary me-3" onClick={() => increment(number)}>
      {number}
    </button>
  );
});
