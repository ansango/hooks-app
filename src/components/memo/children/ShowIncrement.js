import React from "react";
import { log } from "../../../helpers/log";

export const ShowIncrement = React.memo(({ increment }) => {
  log("ShowIncrement", "Render");
  return (
    <button className="btn btn-primary" onClick={() => increment(5)}>
      Incrementar
    </button>
  );
});
