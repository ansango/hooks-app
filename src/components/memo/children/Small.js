import React from "react";
import { log } from "../../../helpers/log";

export const Small = React.memo(({ value }) => {
  log("Small", "Render");
  return <small>{value}</small>;
});
