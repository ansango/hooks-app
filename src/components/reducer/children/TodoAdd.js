import { generateId } from "../../../helpers/generateId";
import { useForm } from "../../../hooks/useForm";

export const TodoAdd = ({ handleAddTodo }) => {
  const [{ description }, handleInputChange, reset] = useForm({
    description: "",
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    if (description.trim().length !== 0) {
      const newTodo = {
        id: generateId(),
        description: description,
        done: false,
      };

      handleAddTodo(newTodo);
      reset();
    }
  };
  return (
    <>
      <h3> Add Todo</h3>
      <hr />
      <form onSubmit={handleSubmit} className="py-2">
        <input
          type="text"
          placeholder="Add a todo"
          name="description"
          autoComplete="off"
          className="form-control mb-3"
          value={description}
          onChange={handleInputChange}
        />
        <button className="btn btn-outline-primary w-100" type="submit">
          Add
        </button>
      </form>
    </>
  );
};
