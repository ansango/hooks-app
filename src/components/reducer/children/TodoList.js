import { TodoListItem } from "./TodoListItem";

export const TodoList = ({ todos, handleDeleteTodo, handleToggleTodo }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, index) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
          index={index}
          handleDeleteTodo={handleDeleteTodo}
          handleToggleTodo={handleToggleTodo}
        />
      ))}
    </ul>
  );
};
