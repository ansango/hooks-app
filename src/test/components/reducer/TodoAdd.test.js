import { TodoAdd } from "../../../components/reducer/children/TodoAdd";
import { shallow } from "enzyme";

describe("Test <TodoAdd/>", () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />);
  test("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should not to call add todo", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });

  test("should call handleAddTodo", () => {
    const value = "Learn Vue";
    wrapper.find("input").simulate("change", {
      target: { value, name: "description" },
    });

    const formSubmit = wrapper.find("form").prop("onSubmit");
    formSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
    expect(handleAddTodo).toHaveBeenCalledWith({
      description: value,
      done: false,
      id: expect.any(String),
    });

    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
