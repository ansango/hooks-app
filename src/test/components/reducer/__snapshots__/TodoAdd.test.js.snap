// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TodoAdd/> should render 1`] = `
<Fragment>
  <h3>
     Add Todo
  </h3>
  <hr />
  <form
    className="py-2"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control mb-3"
      name="description"
      onChange={[Function]}
      placeholder="Add a todo"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary w-100"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
